core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -139,10 +142,19 @@ class LogManager(val logDirs: Array[File],
         recoveryPoints = this.recoveryPointCheckpoints(dir).read
       } catch {
         case e: Exception =>
-          warn("Error occured while reading recovery-point-offset-checkpoint file of directory " + dir, e)
+          warn("Error occurred while reading recovery-point-offset-checkpoint file of directory " + dir, e)
           warn("Resetting the recovery checkpoint to 0")
       }
 
+      var logStartOffsets = Map[TopicPartition, Long]()
+      try {
+        logStartOffsets = this.logStartOffsetCheckpoints(dir).read
+      } catch {
+        case e: Exception =>
+          warn("Error occurred while reading log-start-offset-checkpoint file of directory " + dir, e)
+          warn("Resetting the log-start-offset checkpoint to 0")
```

Comment:
This is kind of similar to what we do with recoveryPoints if its checkpoint file is unreadable. For example, if recoveryPoints checkpoint file is unreadable but `.kafka_cleanshutdown` file exists, we will reset recoveryPoint to `activeSegment.nextOffset` after warning `Resetting the recovery checkpoint to 0`. Here we assume the log-start-offset in the checkpoint file is 0 if checkpoint file is not readable.

On a second thought, I agree it is confusing because we will always set `logStartOffset` to at least base offset of the first second if its checkpoint file is unreadable. I will remove this statement.